### PR TITLE
Bundle deployment: allow providing the path to a local bundle file.

### PR DESCRIPTION
**Pull request by frankban**
(Review request: http://reviews.vapour.ws/r/2759/)

### CODE REVIEW

File: cmd/juju/commands/deploy.go [hunk 1]
```diff
@@ -190,13 +200,31 @@ func (c *DeployCommand) Run(ctx *cmd.Context) error {
...
 	defer csClient.jar.Save()
-
 	repoPath := ctx.AbsPath(c.RepoPath)
-	curl, repo, err := resolveCharmStoreEntityURL(c.CharmName, csClient.params, repoPath, conf)
+
+	// Handle local bundle paths.
+	f, err := os.Open(c.CharmOrBundle)
+	if err == nil {
+		defer f.Close()
+		bundleData, err := charm.ReadBundleData(f)
```
Comment (rogpeppe) [inline cmd/juju/commands/deploy.go]: Perhaps check whether the file is a directory here and produce an appropriate error here?

File: cmd/juju/commands/deploy.go [hunk 2]
```diff
@@ -66,6 +71,14 @@ However, for local charms, when the default-series is not specified in the
 environment, one must specify the series. For example:
   local:precise/mysql
 
+For local bundles the bundle series must be always specified. For example:
```
Comment (rogpeppe) [inline cmd/juju/commands/deploy.go]: This is immediately contradicted by the following paragraph. How about something like:

    Local bundles can be specified either with a local:bundle/<name> URL,
    which is interpreted relative to $CHARM_DIR, or with a direct path to a
    bundle.yaml file. For example, to deploy the bundle in $CHARM_DIR/bundle/openstack,

        juju deploy local:bundle/openstack

    To deploy this using a direct path:

        juju deploy $CHARM_DIR/bundle/openstack/bundle.yaml

?

**Comment by rogpeppe** [comment]
LGTM with a couple of minor suggestions, thanks.

File: cmd/juju/commands/deploy.go (hunk 1 above)
Comment (frankban) [inline cmd/juju/commands/deploy.go]: Done.

File: cmd/juju/commands/deploy.go (hunk 2 above)
Comment (frankban) [inline cmd/juju/commands/deploy.go]: Done.

**Comment by frankban** [comment]
$$merge$$

**Pull request merged**